ocl_cuckaroo/src/trimmer.rs

```diff
@@ -417,6 +458,48 @@ fn find_device(platform: &Platform, selector: Option<usize>) -> ocl::Result<Devi
 	}
 }
 
+fn check_device_compatibility(
+	device: &Device,
+	buffers: &HashMap<String, ClBufferParams>,
+) -> ocl::Result<()> {
+	let max_alloc_size: u64 = get_device_max_alloc_size!(device);
+	let global_memory_size: u64 = get_device_global_mem_size!(device);
+	let mut total_alloc: u64 = 0;
+
+	// Check that no buffer is bigger than the max memory allocation size
+	for (k, v) in buffers {
+		total_alloc += v.size as u64;
+		if v.size as u64 > max_alloc_size {
+			return Err(ocl::Error::from(format!(
+				"Buffer {} is bigger than maximum alloc size ({})",
+				k, max_alloc_size
+			)));
+		}
+	}
+
+	// Check that total buffer allocation does not exceed global memory size
+	if total_alloc > global_memory_size {
```

Comment:
I'm not sure that all buffers are used in a kernel invocation

Reply:
From what I can understand from the code all the buffers are allocated anyway by the Buffer builder function even though as you suggested they might not be used later. It seems that the `recover` function uses `buffer_r` and `buffer_nonces` while the others are used in the `run` function. Should we perform the allocation only when needed?

Also note that `buffer_r` and `buffer_nonces` are pretty small buffers.

Reply:
Yeah, I mean in general, not very future-proof. However chances that we will change the code are small, it's ok to have this check.